internal/core/src/storage/Util.cpp

```diff
@@ -597,9 +598,7 @@ EncodeAndUploadIndexSlice(ChunkManager* chunk_manager,
                           FieldDataMeta field_meta,
                           std::string object_key) {
     // index not use valid_data, so no need to set nullable==true
-    auto field_data = CreateFieldData(DataType::INT8, false);
-    field_data->FillFieldData(buf, batch_size);
-    auto indexData = std::make_shared<IndexData>(field_data);
+    auto indexData = std::make_shared<IndexData>(buf, batch_size);
     indexData->set_index_meta(index_meta);
```

Comment:
cpu usage for indexnode will also benefit from this line, for saving one-time memory-copy